devices/Scd30/Extensions/SerialPortExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿//
+// Copyright (c) 2017 The nanoFramework project contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.IO.Ports
+{
+    internal static class SerialPortExtensions
+    {
+        /// <summary>
+        /// Using <see cref="SerialPort.Read(byte[], int, int)"/> to fetch a specific amount of bytes when they are not
+        /// yet waiting in the buffer causes a timeout. Even if the bytes are received within that time. Instead, this
+        /// helper will use Thread.Sleep() to allow background processing on the serial port. When BytesToRead reaches
+        /// the expected number of bytes, we're done.
+        /// </summary>
+        /// <param name="serialPort">The serial port on which to wait</param>
+        /// <param name="numberOfBytes">The number of bytes to wait for</param>
+        /// <param name="sleepTimeInMs"></param>
+        /// <exception cref="TimeoutException"></exception>
+        internal static void WaitForData(this SerialPort serialPort, int numberOfBytes, int sleepTimeInMs = 5)
+        {
+            if (serialPort.ReadBufferSize < numberOfBytes)
+                throw new ArgumentOutOfRangeException(nameof(numberOfBytes), $"The SerialPort.ReadBufferSize is too small ({serialPort.ReadBufferSize}) to facilitate waiting for {numberOfBytes} bytes");
```

Comment:
That will be catch by the linter when you'll add it, we use explicit brackets for clarity.